lisa/sut_orchestrator/qemu/platform.py

```diff
@@ -360,16 +404,20 @@ def _delete_nodes(
             except Exception as ex:
                 log.warning(f"console log delete failed. {ex}")
 
-            # Delete the files created for the VM.
-            try:
-                os.remove(node_context.os_disk_file_path)
-            except Exception as ex:
-                log.warning(f"OS disk delete failed. {ex}")
+            if self.qemu_platform_runbook.is_host_remote():
+                vm_dir = os.path.dirname(node_context.os_disk_file_path)
+                self.host_node.shell.remove(PurePath(vm_dir), True)
+            else:
+                # Delete the files created for the VM.
+                try:
+                    os.remove(node_context.os_disk_file_path)
```

Comment:
If `host_node` is set to `local_host` for the local case, then this can become `host_node.shell.remove`.